src/streams.lisp

```diff
@@ -500,14 +559,22 @@ may be associated with the passphrase PASSWORD."
     (cffi:foreign-string-to-lisp (cffi:mem-ref ptr :pointer)
                                  :count (cffi:mem-ref len :int))))
 
-#+openmcl
+#+(or openmcl clozure-common-lisp)
 (defmethod stream-deadline ((stream ccl::basic-stream))
   (ccl::ioblock-deadline (ccl::stream-ioblock stream t)))
-#+openmcl
+
+(defmethod stream-deadline ((stream usocket:stream-usocket))
+  (+ (get-internal-real-time)
+     (* internal-time-units-per-second (usocket:socket-option stream :receive-timeout))))
+
+#+sbcl
+(defmethod stream-deadline ((stream sb-sys:fd-stream))
+  (+ (get-internal-real-time)
+     (* internal-time-units-per-second (sb-impl::fd-stream-timeout stream))))
+
```

Comment:
I would not want to support portability code for stream timeout extraction in cl+ssl. Not only because that's a pile of conventionalized impl-specific code, but also because it raises some inconvenient questions. Like why do use use `:receive-timeout` for usocket streams, and not `:send-timeout`? What if the lisp implementation does not support stream timeouts, or supports only read timeouts?

It's better to leave the automatic extraction of timeout from stream for CCL backward compatibility support only. 

The users will have to pass the timeouts explicitly as parameters - if they know how to set the timeout in a platform specific way, it should be easy for them to pass the same values to cl+ssl. 

BTW, maybe a separate small portability library could be created for lisp stream timeouts (probably using https://github.com/edicl/hunchentoot/blob/master/set-timeouts.lisp as the starting point). But that's outside of the cl+ssl scope.

Reply:
Agreed, this was an unnecessary feature addition.